deepchem/molnet/load_function/kt_dataset.py

```diff
@@ -0,0 +1,210 @@
+"""
+Loading Kamlet-Taft (KT) parameter dataset 
+"""
+
+import os
+import logging
+import deepchem
+from deepchem.feat import Featurizer
+from deepchem.trans import Transformer
+from deepchem.splits.splitters import Splitter
+from deepchem.molnet.defaults import get_defaults
+
+from typing import List, Tuple, Dict, Optional
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_DIR = deepchem.utils.get_data_dir()
+KTDATASET_URL = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/KTparameterDataset.zip"
+KTDATASET_CSV_URL = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/KTparameterDataset.csv"
+
+# dict of accepted featurizers for the KT dataset
+DEFAULT_FEATURIZERS = get_defaults("feat")
+
+# Names of supported featurizers
+# ktdataset_featurizers = ["RawFeaturizer", "SmilesToImage", "SmilesToSeq", "WeaveFeaturizer"]
+ktdataset_featurizers = ['ECFP', 'GraphConv', 'Weave', 'Raw', 'AdjacencyConv']
+DEFAULT_FEATURIZERS = {k: DEFAULT_FEATURIZERS[k] for k in ktdataset_featurizers}
+
+# dict of accepted transformers
+DEFAULT_TRANSFORMERS = get_defaults("trans")
+
+# dict of accepted splitters
+DEFAULT_SPLITTERS = get_defaults("splits")
+
+# names of supported splitters
+ktdataset_splitters = [
+    'index', 'random', 'scaffold', 'butina', 'task', 'stratified'
+]
+DEFAULT_SPLITTERS = {k: DEFAULT_SPLITTERS[k] for k in ktdataset_splitters}
+
+
+def load_kt_dataset(
+    featurizer: Featurizer = DEFAULT_FEATURIZERS['RawFeaturizer'],
+    transformers: List[Transformer] = [
+        DEFAULT_TRANSFORMERS['PowerTransformer']
+    ],
+    splitter: Splitter = DEFAULT_SPLITTERS['RandomSplitter'],
+    reload: bool = True,
+    data_dir: Optional[str] = None,
+    save_dir: Optional[str] = None,
+    featurizer_kwargs: Dict[str, object] = {},
+    splitter_kwargs: Dict[str, object] = {},
+    transformer_kwargs: Dict[str, Dict[str, object]] = {},
+    **kwargs) -> (Tuple[List, Tuple, List]):
+
+"""
+
+  KT parameters are scales to measure and quantify the Lewis acidity and basicity of molecules. The parameters are obtained through Nuclear Magnetic Resonance (NMR) spectra. Random splitting is recommended for this dataset.
+  The raw data csv file contains columns below:
+  Please refer to https://arxiv.org/pdf/2008.08078 (2020) for which the dataset was hand curated. Benchmark training is also present in this work. Accuracy levels (rmse) reported here are of the order 0.01  for alpha and beta on unseen data. 
```

Comment:
Could you add this to the references section? Here's an example (taken from the numpy docs guide https://numpydoc.readthedocs.io/en/latest/format.html)

```
Please refer to reference [1]_ 

References
-----------
.. [1] O. McNoleg, "The integration of GIS, remote sensing,
   expert systems and adaptive co-kriging for environmental habitat
   modelling of the Highland Haggis using object-oriented, fuzzy-logic
   and neural-network techniques," Computers & Geosciences, vol. 22,
   pp. 585-588, 1996.
```

Reply:
Ah just saw you already have a references section. The arxiv paper can be added there :)

Reply:
Thanks a lot, Bharath. I have updated the file accordingly.